axelor-account/src/main/java/com/axelor/apps/account/service/payment/invoice/payment/InvoicePaymentToolServiceImpl.java

```diff
@@ -248,15 +248,29 @@ public void checkConditionBeforeSave(InvoicePayment invoicePayment) throws Axelo
 
   @Override
   public BigDecimal getPayableAmount(
-      List<InvoiceTerm> invoiceTermList, LocalDate date, boolean manualChange) {
-    return invoiceTermList.stream()
-        .map(
-            it ->
-                manualChange
-                    ? invoiceTermService.getAmountRemaining(it, date)
-                    : it.getAmountRemaining())
-        .reduce(BigDecimal::add)
-        .orElse(BigDecimal.ZERO);
+      List<InvoiceTerm> invoiceTermList,
+      LocalDate date,
+      boolean manualChange,
+      Currency targetCurrency)
+      throws AxelorException {
+    Currency startCurrency = invoiceTermList.get(0).getCurrency();
+    BigDecimal amount =
+        invoiceTermList.stream()
+            .map(
+                it ->
+                    manualChange
+                        ? invoiceTermService.getAmountRemaining(it, date)
+                        : it.getAmountRemaining())
+            .reduce(BigDecimal::add)
+            .orElse(BigDecimal.ZERO);
+    if (CollectionUtils.isNotEmpty(invoiceTermList)) {
+      if (!startCurrency.equals(targetCurrency)) {
+        amount =
+            currencyService.getAmountCurrencyConvertedAtDate(
+                startCurrency, targetCurrency, amount, date);
+      }
+    }
+    return amount;
```

Comment:
We never redo a conversion, we have 2 fields on an InvoiceTerm to manage multi currency:
amountRemaining = the remaining amount in the currency on the InvoiceTerm
companyAmountRemaining = the remaining amount in the currency on the company

That's all you need, so instead of doing the conversion, you need to get the right remaining amount, amountRemaining or companyAmountRemaining depending on the payment's currency.
And the payment's currency will always be equal to the InvoiceTerm currency or the company currency because in AOS, per example we don't allow to do a payment in £ if the InvoiceTerm currency is $ and the company currency is €.